### PR TITLE
boot2docker -> docker-machine in meso-docker

### PR DESCRIPTION
**Pull request by cmluciano**
This is related to the cleanup in #17508.

This PR also contains a commit to add VMware instructions for boot2docker.
I can split this into two PRs if anyone wants me to.

### CODE REVIEW

**Comment by jdef** [comment]
@karlkfi PTAL

File: docs/getting-started-guides/mesos-docker.md [hunk 1]
```diff
@@ -118,17 +118,29 @@ In order to build Kubernetes, the current user must be in a docker group with su
 See the docker docs for [instructions](https://docs.docker.com/installation/ubuntulinux/#create-a-docker-group).
 
 
-#### Boot2Docker Config (Mac)
+#### docker-machine Config (Mac)
```
Comment (karlkfi) [inline docs/getting-started-guides/mesos-docker.md]: `Docker Machine` is the name. docker-machine is the cmd.

Comment (karlkfi) [inline docs/getting-started-guides/mesos-docker.md]: Need to insert a step 1 that creates a docker-machine vm. 
Ex:

```
docker-machine create --driver=virtualbox --virtualbox-cpu-count=4 --virtualbox-disk-size=102400 --virtualbox-memory=4096 dev
```

File: docs/getting-started-guides/mesos-docker.md [hunk 2]
```diff
@@ -118,17 +118,29 @@ In order to build Kubernetes, the current user must be in a docker group with su
...
 
-If on a mac using boot2docker, the following steps will make the docker IPs (in the virtualbox VM) reachable from the
+If on a mac using docker-machine, the following steps will make the docker IPs (in the virtualbox/vmware VM) reachable from the
 host machine (mac).
 
 1. Set the VM's host-only network to "promiscuous mode":
```
Comment (karlkfi) [inline docs/getting-started-guides/mesos-docker.md]: Separate the promiscuous mode step from the create step, please.

File: docs/getting-started-guides/mesos-docker.md [hunk 3]
```diff
@@ -78,12 +78,14 @@ Required:
...
 - [Docker CLI](https://docs.docker.com/) - container management command line client
 - [Docker Engine](https://docs.docker.com/) - container management daemon
-  - On Mac, use [Boot2Docker](http://boot2docker.io/) or [Docker Machine](https://docs.docker.com/machine/install-machine/)
+  - On Mac, use [Docker Machine](https://docs.docker.com/machine/install-machine/)
 - [Docker Compose](https://docs.docker.com/compose/install/) - multi-container application orchestration
 
 Optional:
 - [Virtual Box](https://www.virtualbox.org/wiki/Downloads) - x86 hardware virtualizer
-  - Required by Boot2Docker and Docker Machine
+  - Required by Docker Machine
```
Comment (karlkfi) [inline docs/getting-started-guides/mesos-docker.md]: "Free x86 virtualization engine with a Docker Machine driver"

File: docs/getting-started-guides/mesos-docker.md [hunk 4]
```diff
@@ -78,12 +78,14 @@ Required:
...
+- [VMware Fusion](https://www.vmware.com/products/fusion) - x86 hardware virtualizer
+  - Can be used in place of Virtualbox by Docker Machine
```
Comment (karlkfi) [inline docs/getting-started-guides/mesos-docker.md]: Proprietary x86 virtualization engine with a Docker Machine driver

File: docs/getting-started-guides/mesos-docker.md [hunk 5]
```diff
@@ -198,9 +210,9 @@ host machine (mac).
     If you delete the `MESOS_RESOURCES` environment variables, the resource amounts will be auto-detected based on the host resources, which will over-provision by > 2x.
 
     If the configured resources are not available on the host, you may want to increase the resources available to Docker Engine.
-    You may have to increase you VM disk, memory, or cpu allocation in VirtualBox,
-    [Docker Machine](https://docs.docker.com/machine/#oracle-virtualbox), or
-    [Boot2Docker](https://ryanfb.github.io/etc/2015/01/28/increasing_boot2docker_allocations_on_os_x.html).
+    You may have to increase you VM disk, memory, or cpu allocation
+    [Docker Machine Virtualbox](https://docs.docker.com/machine/drivers/virtualbox)
+    [Docker Machine VMware](https://docs.docker.com/machine/drivers/vm-fusion/)
```
Comment (karlkfi) [inline docs/getting-started-guides/mesos-docker.md]: You may have to increase the Docker Machine VM's disk, memory, or cpu allocation. See the Docker Machine docs for details ([VirtualBox](https://docs.docker.com/machine/drivers/virtualbox), [VMware](https://docs.docker.com/machine/drivers/vm-fusion/)).

**Comment by cmluciano** [comment]
Iterated on code review

**Comment by karlkfi** [comment]
The changes look great. I'm just hesitant to merge this into the doc until vmware actually works...

We probably have to just put this on ice until https://github.com/mesosphere/kubernetes-mesos/issues/630 is resolved. Otherwise more people will run into in, thinking it's supported.

**Comment by cmluciano** [comment]
Should I split out those changes into a separate PR?

**Comment by karlkfi** [comment]
Yeah, that's probably a good idea for now. Then we can merge the docker-machine ones.

**Comment by karlkfi** [comment]
You'll need to update the toc by running `run hack/update-generated-docs.sh`. You probably want to read the devel docs and set up the commit hooks to make sure they all pass.

You'll also need to sign the CLA and make sure your git commit uses the same email you signed with.

Other than that it looks good to me.

**Comment by karlkfi** [comment]
Looks like this still needs a signed CLA (or the correct email on the commit) and a rebase. Other than that it lgtm, and I'd be happy to merge it.

**Comment by cmluciano** [comment]
I'm not sure how this got closed.

**Comment by karlkfi** [comment]
Looks like it auto-closed because the branch has no commits on it or was deleted.

**Pull request closed without merging**